sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from "@nestjs/common";
+import { RecordDataModelService } from "../../database/data.model.service";
+import { Connection, SelectQueryBuilder } from "typeorm";
+import {
+  StudentAppealRequest,
+  StudentAppealStatus,
+} from "../../database/entities";
+
+@Injectable()
+export class StudentAppealRequestsService extends RecordDataModelService<StudentAppealRequest> {
+  constructor(connection: Connection) {
+    super(connection.getRepository(StudentAppealRequest));
+  }
+  /**
+   * Creates a 'select' query that could be used in an 'exists' or
+   * 'not exists' where clause.
+   * @param appealStatus student request status that to be searched for.
+   * @param isEqual, this flag decides how the passed status should checked,
+   * eg, if isEqual is true, then the logic will check if the passed status
+   * is equal to appealStatus.
+   * if isEqual is false, then the logic will check if the passed status
+   * is not equal to appealStatus.
+   * by default isEqual is true.
+   * @returns 'select' query that could be used in an 'exists' or
+   * 'not exists'.
+   */
+  getExistsAppeals(
```

Comment:
IMO the method can be renamed as getAppealBystatus or getAppealByStatusAndOperator. I believe our intention is to pass the status and operator(Equals, NotEqual)

Reply:
as in comment it's used in exist and non exists, we have a lot of examples for subquery, please check and let me know 
![image](https://user-images.githubusercontent.com/77353155/160464943-2a59c11d-77d7-44e5-90ce-39762d978810.png)

Reply:
If the method will be changed I would prefer the first suggested option `getAppealByStatus`.

Reply:
updated to `appealsByStatusQueryObject`
![image](https://user-images.githubusercontent.com/77353155/160470698-7de55ce2-26be-4ef9-8381-d907d0ee75e9.png)